super_editor/lib/src/infrastructure/composable_text.dart

```
@@ -0,0 +1,55 @@
+import 'package:flutter/rendering.dart';
+
+/// Contract for widgets that include editable text.
+///
+/// Examples: paragraphs, list items, images with captions.
+///
+/// The text positions accepted by a [TextComposable] are [dynamic]
+/// rather than [TextPosition]s because a use-case might include
+/// complex text composition, like tables, which might choose to
+/// index positions based on cell IDs, or row and column indices.
+abstract class TextComposable {
+  /// Returns a [TextSelection] that encompasses the entire word
+  /// found at the given [textPosition].
+  ///
+  /// Throws an exception if [textPosition] is not the right type
+  /// for this [TextComposable].
+  TextSelection getWordSelectionAt(dynamic textPosition);
```

I am not so sure if we should throw an exception without a proper definition of allowed types here. Can we put something to class documentation? I know we already have possibilities but if we let the user decide the type, then I don't think that throwing non-allowed types should be their concern more than ours.

---

That was done so that something like a table could still be considered a `TextComposable`. It's mentioned in the class docs:

```
/// The text positions accepted by a [TextComposable] are [dynamic]
/// rather than [TextPosition]s because a use-case might include
/// complex text composition, like tables, which might choose to
/// index positions based on cell IDs, or row and column indices.
```